cdap-common/src/main/java/co/cask/cdap/common/HttpExceptionHandler.java

```diff
@@ -37,32 +37,13 @@
 
   @Override
   public void handle(Throwable t, HttpRequest request, HttpResponder responder) {
+    // Check if the exception is caused by Service being unavailable: this will happen during master startup
     if (Iterables.size(Iterables.filter(Throwables.getCausalChain(t), ServiceUnavailableException.class)) > 0) {
+      // no need to log ServiceUnavailableException because at master startup we are waiting for services to come up
       responder.sendString(HttpResponseStatus.SERVICE_UNAVAILABLE, t.getMessage());
-    } else if (t instanceof BadRequestException) {
+    } else if (t instanceof HttpResponseException) {
       logWithTrace(request, t);
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, t.getMessage());
-    } else if (t instanceof ConflictException) {
-      logWithTrace(request, t);
-      responder.sendString(HttpResponseStatus.CONFLICT, t.getMessage());
-    } else if (t instanceof NotFoundException) {
-      logWithTrace(request, t);
-      responder.sendString(HttpResponseStatus.NOT_FOUND, t.getMessage());
-    } else if (t instanceof NotImplementedException) {
-      logWithTrace(request, t);
-      responder.sendString(HttpResponseStatus.NOT_IMPLEMENTED, t.getMessage());
-    } else if (t instanceof MethodNotAllowedException) {
-      logWithTrace(request, t);
-      responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
-    } else if (t instanceof UnauthenticatedException) {
-      logWithTrace(request, t);
-      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
-    } else if (t instanceof UnauthorizedException) {
```

Comment:
where is it used? any way it can be removed? ok for now, but perhaps we should think about it.